pkg/api/types.go

```diff
@@ -138,6 +138,17 @@ type ObjectMeta struct {
 	// objects.  Annotation keys have the same formatting restrictions as Label keys. See the
 	// comments on Labels for details.
 	Annotations map[string]string `json:"annotations,omitempty"`
+
+	// List of objects depended by this object. If ALL objects in the list have
+	// been deleted, this object will be garbage collected.
+	OwnerReferences []ObjectReference `json:"ownerReferences,omitempty"`
+
+	// Must be empty before the object is deleted from the registry. The map key
+	// is an identifier for the responsible component that will remove the entry
+	// from the map, the value is an ad hoc, explanatory sentence (e.g.,
+	// "orphaning finalizer"). If the ObjectMeta.DeletionTimestamp is non-nil,
+	// entries in this map can only be removed.
+	Finalizers map[string]string `json:"finalizers,omitempty"`
```

Comment:
Can this be just a list, with the understanding that "foo" gets the default "foo" finalizer and "foo=special" gets the "special" finalizer of class foo?

I don't think having an "ad hoc explanatory sentence" stored for every object is a great use of space.